include/Patch.h

```diff
@@ -1101,5 +1101,83 @@ struct patch_t
 }; // struct patch_t
 
 
+class NonCopyable
+{
+  protected:
+    NonCopyable() {}
+    ~NonCopyable() {}
+  private: 
+    NonCopyable(const NonCopyable &);
+    NonCopyable& operator=(const NonCopyable &);
+}; // class NonCopyable
+
+struct LB_GlobalPatch
+{
+   int    corner[3];
+   long   sibling[26];
+   long   father;
+   long   son;
```

Comment:
Yes, you are right about this. I was not consistent. I personally do not have a preference here and the only case where it would cause a problem is if we had a simulation with more than 2^32 - 1 = 2,147,483,647. I doubt that is feasible. So, I guess that an integer should be safe here. Should I also change the corresponding long datatypes in YT?

Reply:
I have changed the datatype to int in my code now. With this all your comments should have been fixed. I have tested the code in SERIAL mode (HDF5 and libYT) and will now run it with MPI.

Reply:
@cindytsai Did you test it and find out you had to use the long datatype for GIDs?

Reply:
@KunkelAlexander `int` is definitely OK for GAMER since each patch already contains at least `8^3` cells. `libyt` is more general-purpose so I'm not sure. But as long as this PR is still compatible with `libyt` it should be fine (i.e., make sure that we _copy_ GID instead of passing the entire GID-related _array_ to `libyt`.

Reply:
@hyschive Got it, then I will leave the `long` types in the `libyt`-related code untouched. I also realised that `libyt` expects function pointers to have `long` types as arguments.